assets/js/modules/adsense/components/settings/AdBlockingRecoveryApp/SetupMain.test.js

```diff
@@ -0,0 +1,408 @@
+				.getAdminURL( 'googlesitekit-dashboard' );
+
+			// eslint-disable-next-line require-await
+			await act( async () => {
+				fireEvent.click( getByRole( 'button', { name: /cancel/i } ) );
+			} );
+
+			expect( mockTrackEvent ).toHaveBeenCalledWith(
+				'adBlockingRecovery_adsense-abr',
+				'cancel_setup',
+				'on_place_tag_step'
+			);
+
+			expect( global.location.assign ).toHaveBeenCalled();
+			expect( global.location.assign ).toHaveBeenCalledWith(
+				dashboardURL
+			);
+		} );
+	} );
+
+	describe( 'Create Message step', () => {
+		beforeEach( async () => {
+			muteFetch(
+				new RegExp(
+					'^/google-site-kit/v1/modules/adsense/data/sync-ad-blocking-recovery-tags'
+				)
+			);
+
+			muteFetch(
+				new RegExp(
+					'^/google-site-kit/v1/modules/adsense/data/settings'
+				)
+			);
+
+			// eslint-disable-next-line require-await
+			await act( async () => {
+				fireEvent.click(
+					getByRole( 'button', { name: /enable message/i } )
+				);
+			} );
+		} );
+
+		it( 'Should fire the event tracking when rendered', () => {
+			expect( container ).toHaveTextContent(
+				'Create and publish an ad blocking recovery message in AdSense'
+			);
+
+			expect( mockTrackEvent ).toHaveBeenCalledWith(
+				'adBlockingRecovery_adsense-abr',
+				'setup_create_message'
+			);
+		} );
+
+		it( 'should render the `Final (My message is ready)` step when `Create message` button is clicked', async () => {
+			// eslint-disable-next-line require-await
+			await act( async () => {
+				fireEvent.click(
+					getByRole( 'button', { name: /create message/i } )
+				);
+			} );
+
+			expect( container ).toHaveTextContent(
+				'Ad blocking recovery only works if you’ve created and published your message in AdSense'
+			);
+
+			expect( mockTrackEvent ).toHaveBeenCalledWith(
+				'adBlockingRecovery_adsense-abr',
+				'create_message',
+				'primary_cta'
+			);
+
+			expect( container ).toHaveTextContent(
+				'Ad blocking recovery only works if you’ve created and published your message in AdSense'
+			);
+		} );
+
+		it( 'should return to dashboard when `Cancel` button is clicked', async () => {
+			const dashboardURL = registry
+				.select( CORE_SITE )
+				.getAdminURL( 'googlesitekit-dashboard' );
+
+			fetchMock.postOnce(
+				/^\/google-site-kit\/v1\/modules\/adsense\/data\/settings/,
+				{ body: { success: true }, status: 200 }
+			);
+
+			// eslint-disable-next-line require-await
+			await act( async () => {
+				fireEvent.click( getByRole( 'button', { name: /cancel/i } ) );
+			} );
+
+			expect( mockTrackEvent ).toHaveBeenCalledWith(
+				'adBlockingRecovery_adsense-abr',
+				'cancel_setup',
+				'on_create_message_step'
+			);
+
+			// Rollback Ad Blocking Recovery Settings.
+			expect( JSON.parse( fetchMock.lastOptions().body ) ).toStrictEqual(
+				{
+					data: {
+						adBlockingRecoverySetupStatus: '',
+						useAdBlockingRecoveryErrorSnippet: false,
+						useAdBlockingRecoverySnippet: false,
+					},
+				}
+			);
```

Comment:
This seems a bit fragile to make assertions about the last fetch – why not make the same assertions about the state in the data store?

Reply:
We can't assert on registry because output the `getSettings()` and the `get*` selectors will be replaced by the mock response of `google-site-kit/v1/modules/adsense/data/settings`. We can't `freezeFetch` here as well.